lib/models/applicant.rb

```diff
@@ -23,4 +26,11 @@ def prefers_interest
     !prefers_nearby
   end
   alias_method :prefers_interest?, :prefers_interest
+
+  private 
+
+    def compute_grid_id
+      @grid = Grid.intersecting_grid(location: location)
+      self.grid_id = @grid.g250m_id if @grid
```

Comment:
The record is not saved if there is not @grid. Applicant model validates the presence of grid_id.